dl1_data_handler/reader.py

```diff
@@ -1574,9 +1641,10 @@ def __init__(
         )
 
         # Read the readout length from the first file
+        data_group = getattr(self.files[self.first_file].root, self.data_level)
```

Comment:
why here we have `self.data_level` and above `data_level = "r1"`? I think each reader subclass should hold a datalevel attribute from the enum (see comment above). Then you can your desired string `self.data_level.name.lower()`.